hydra-module.nix

```diff
@@ -175,65 +246,733 @@ in
         default = false;
         description = ''
           Whether to use binary caches for downloading store paths. Note that
-          binary substitutions trigger (a potentially large number of) additional
+          binary substitutions trigger a potentially large number of additional
           HTTP requests that slow down the queue monitor thread significantly.
           Also, this Hydra instance will serve those downloaded store paths to
           its users with its own signature attached as if it had built them
           itself, so don't enable this feature unless your active binary caches
           are absolute trustworthy.
         '';
       };
-    };
 
-  };
+      googleClientID = mkOption {
```

Comment:
Is this `googleClientID` a secret or is it safe to store in the configuration?

Reply:
I think it is a secret.

Reply:
This is not a bad idea. Will investigate.

Reply:
+1. I think it's currently impossible to configure Hydra with the current NixOS module without putting e.g. github auth tokens in the nix store.

Reply:
@ElvishJerricco: I have an example NixOps deploy of Hydra that patches Hydra to fix this.  You can see the deploy here:

https://github.com/dhall-lang/dhall-lang/tree/master/nixops

... and the patch here:

https://github.com/dhall-lang/dhall-lang/blob/master/nixops/hydra.patch